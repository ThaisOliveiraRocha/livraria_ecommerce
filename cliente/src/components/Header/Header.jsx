import React, { useState } from "react";
import PropTypes from "prop-types";
import {
  Container,
  Image,
  CartContainer,
  CartImage,
  QuantidadeItens,
  HomeComponent
} from "./Header.styles";

import { connect } from "react-redux";

import Button from "../Button";
import Titulo from "../Title";
import Texto from "../Text";
import CartModal from "../CartModal";
import { withRouter } from "react-router-dom";

const totalCart = carrinho => {
  const total = carrinho.reduce((aux, livro) => {
    aux = aux + livro.qtd;
    return aux;
  }, 0);
  return total;
};

const Header = ({ carrinho, history }) => {
  const [showModal, setModalStatus] = useState(false);
  return (
    <Container>
      <HomeComponent onClick={() => history.push("/")}>
        <Image src={require("../../assets/images/book_novo.svg")} />
        <Titulo font="28px" color="#ffffff" grow="1">
          Livraria on-line
        </Titulo>
      </HomeComponent>
      <CartContainer onClick={() => setModalStatus(!showModal)}>
        <Button>
          <CartImage
            src={require("../../assets/images/carrinho_mercado.png")}
          />
          <Texto>Carrinho</Texto>
        </Button>
        {carrinho.length > 0 && (
          <QuantidadeItens>{totalCart(carrinho)}</QuantidadeItens>
        )}
      </CartContainer>
      {showModal && (
        <CartModal carrinho={carrinho} onClose={() => setModalStatus(false)} />
      )}
    </Container>
  );
};

Header.propTypes = {
  // bla: PropTypes.string,
};

const mapStateToProps = state => ({
  carrinho: state.livraria.carrinho
});

export default withRouter(connect(mapStateToProps)(Header));
