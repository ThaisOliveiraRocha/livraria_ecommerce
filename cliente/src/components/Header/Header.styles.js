import styled from "styled-components";

export const Container = styled.div`
  display: flex;
  width: 100%;
  height: 70px;
  padding: 20px 25px 20px 25px;
  justify-content: center;
  align-items: center;
  background-color: #2a7886;
  position: fixed;
  top: 0;
  z-index: 2;
`;

export const Image = styled.img`
  width: 40px;
  height: auto;
  margin-right: 5px;
`;

export const CartImage = styled.div`
  width: 30px;
  height: auto;
`;

export const QuantidadeItens = styled.span`
  border-radius: 50%;
  background-color: red;
  font-size: 12px;
  width: 15px;
  height: 15px;

  display: flex;
  align-items: center;
  justify-content: center;
  position: absolute;
  top: 8px;
  left: 45px;
`;

export const CartButtonComponent = styled.span`
  display: flex;
  align-items: center;
`;

export const HomeComponent = styled.div`
  display: flex;
  align-items: center;
  justify-content: flex-start;
  width: 50%;
  flex-grow: 1;
`;

export const CartContainer = styled.div`
  width: 25%;
  justify-content: flex-end;
  display: flex;
  align-items: center;
`;

export const UserComponent = styled.div`
  width: 25%;
  justify-content: center;
  display: flex;
  align-items: center;
`;

export const ButtonsComponent = styled.div`
  width: 50%;
  display: flex;
  align-items: center;
  justify-content: flex-end;
`;
