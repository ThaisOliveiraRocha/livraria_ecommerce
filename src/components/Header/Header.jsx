import React, { useState } from "react";
import PropTypes from "prop-types";
import {
  Container,
  Image,
  CartContainer,
  CartImage,
  QuantidadeItens
} from "./Header.styles";

import Button from "../../components/Button";
import Titulo from "../../components/Title";
import CartModal from "../CartModal";

const Header = props => { 
  const [showModal, setModalStatus] = useState(false);
  return (
  <Container>
    <Image src={require("../../assets/images/book_novo.svg")} />
    <Titulo font="28px" color="#ffffff" grow="1">
      Livraria on-line
    </Titulo>
    <CartContainer onClick={() => setModalStatus(!showModal)}>
      <Button>
        <CartImage src={require("../../assets/images/carrinho_mercado.png")} />
        Carrinho
      </Button>
      <QuantidadeItens>1</QuantidadeItens>
    </CartContainer>
    {showModal && <CartModal />}
  </Container>
)};

Header.propTypes = {
  // bla: PropTypes.string,
};

Header.defaultProps = {
  // bla: 'test',
};

export default Header;
