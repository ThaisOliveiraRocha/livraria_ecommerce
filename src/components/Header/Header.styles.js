import styled from "styled-components";

export const Container = styled.div`
  display: flex;
  width: 100%;
  height: 90px;
  padding: 10px;

  align-items: center;
  background-color: #003D99;
`;

export const Image = styled.img`
  width: 50px;
  height: auto;
  margin-right: 5px;
`;

export const Titulo = styled.span`
  flex-grow: 1;
  color: #ffffff;
  font-size: 26px;
  font-weight: 700;
`;

export const CartContainer = styled.div`
  display: flex;
  color: #ffffff;
  align-items: center;

  margin-right: 15px;
`;

export const CartImage = styled.img`
  width: 30px;
  height: auto;
  margin-right: 10px;
`;

export const TituloCarrinho = styled.span`
  flex-grow: 1;
  color: #ffffff;
  font-size: 18px;
  font-weight: 300;
`;

export const QuantidadeItens = styled.span`
  border-radius: 50%;
  background-color: red;
  font-weight: 300;
  width: 20px;
  height: 20px;

  display: flex;
  align-items: center;
  justify-content: center;

  margin-left: 10px;
`;
