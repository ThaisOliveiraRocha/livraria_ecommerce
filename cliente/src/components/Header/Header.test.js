import React from "react";
import { shallow } from "enzyme";
import Header from "./Header";

describe("<Header />", () => {
  it("should render without crashing", () => {
    shallow(<Header />);
  });
  it("should render basic infos", () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();
  });
  
});
