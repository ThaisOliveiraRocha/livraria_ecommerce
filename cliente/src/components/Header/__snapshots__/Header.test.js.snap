// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render basic infos 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
